tool/ood-gen/lib/academic_institution.ml

```diff
@@ -39,7 +39,9 @@ let of_metadata m = of_metadata m ~slug:(Utils.slugify m.name)
 
 let decode (_, (head, body_md)) =
   let metadata = metadata_of_yaml head in
-  let body_html = Omd.of_string body_md |> Omd.to_html in
+  let body_html =
+    Cmarkit.Doc.of_string ~strict:true body_md |> Cmarkit_html.of_doc ~safe:true
```

Comment:
Because Hilite injects some HTML into the markdown document we need to set `safe` to `false`. Provided you trust `hilite`'s HTML (and any other raw HTML that might be in your markdown file) this should be totally "safe" :))

This is why it wasn't rendering the code blocks.

```suggestion
    Cmarkit.Doc.of_string ~strict:true body_md |> Cmarkit_html.of_doc ~safe:false
```

Reply:
Thank you Patrick, that's a good pointer! Indeed, we trust our own HTML from the markdown files here.

Reply:
Thank you Patrick, I'll try this now 😊

Reply:
@patricoferris I tried this now and it rendered the block, but this time the text that's supposed to be highlighted was omitted 

**Output**
<img width="1277" alt="Screenshot 2023-10-20 at 20 59 17" src="https://github.com/ocaml/ocaml.org/assets/64274826/62b2315d-75b3-4697-aa0f-ea21d0f1f4ef">

**Expected output**
<img width="1317" alt="Screenshot 2023-10-20 at 21 09 24" src="https://github.com/ocaml/ocaml.org/assets/64274826/b74dca7b-d433-46c7-a698-0802ee2e681e">

Reply:
I think you are using a slightly older commit on that branch, I think I fixed this in https://github.com/patricoferris/hilite/pull/11/commits/87bd6ad048894ced54a4613d4d4cad951de19669

Reply:
This works now 😅. Thank you @patricoferris ❤️

Reply:
Oh... wait! It doesn't work on diffs 😪 

**Expected output**
<img width="1270" alt="Screenshot 2023-10-21 at 12 05 47" src="https://github.com/ocaml/ocaml.org/assets/64274826/e5d4d251-b036-4119-94b7-dc77c90101a5">

**output**
<img width="1252" alt="Screenshot 2023-10-21 at 12 06 00" src="https://github.com/ocaml/ocaml.org/assets/64274826/b3815d3c-cc3d-4209-8891-6d1bf3eece25">

Reply:
@oyenuga17 that's great progress! We're working with an unreleased upstream patch here, so what you're doing here is helping a lot. :rocket: Looks like another question for @patricoferris.

Reply:
I thought I replied here, but apparently I didn't! Yes @oyenuga17 this has been massively helpful, I needed to rebase the cmarkit PR to pick up the support for diff. Can you try https://github.com/patricoferris/hilite/pull/11/commits/c35d31a3184de85bae434aade62cee90840c2db4

Reply:
Thank you Patrick. I'll try this and give you feedback 😊

Reply:
Thank you @patricoferris, support for diffs works now... I can now proceed with this pr 😊 🙏

Reply:
@sabine I'm a bit stuck on this issue. I took some time to learn the ocaml syntax but it's still quite difficult to replace omd with cmarkit in exercise.ml, tutorial.ml and workshop.ml where omd was heavily used.

Please I need some guidance on these ones, maybe an example of where omd was replaced with cmarkit 😅 🙏

Reply:
@oyenuga17 I think the hardest part in migrating from `omd` to `cmarkit` is probably recreating the table-of-contents generation in `tutorial.ml`. When we have that, we can use this in `workshop.ml` as well.

I would start from scratch here, looking at the Cmarkit docs to see how to traverse the document. I.e. you're given a `Cmarkit.Doc`, and now you need to look at the content block and do whatever is needed to build the table of contents hierarchy.

Btw, the current implementation has a "bug": when the section hierarchy has gaps, e.g. a h1 followed by a h3, there's going to be some entries missing in the generated table of contents.

Either way, to see that this is the only task left to upgrade to cmarkit, you could put a placeholder value and a FIXME comment for the table-of-contents generation.

The splitting done in `exercise.ml` is easier than the table of contents generation, so I think you might be able to figure that out by following the definitions from https://ocaml.org/p/cmarkit/latest/doc/Cmarkit/Doc/index.html to see what the different cases to match on are.

Maybe we want to schedule some pair programming on this. If so, do reach out on Discord (my username is .sabine.).

Reply:
That would be great @sabine... Thank you 😍